docker/client.py

```diff
@@ -780,7 +780,7 @@ def start(self, container, binds=None, port_bindings=None, lxc_conf=None,
             publish_all_ports=publish_all_ports, links=links, dns=dns,
             privileged=privileged, dns_search=dns_search, cap_add=cap_add,
             cap_drop=cap_drop, volumes_from=volumes_from, devices=devices,
-            network_mode=network_mode, restart_policy=restart_policy,
+            network_mode=network_mode or '', restart_policy=restart_policy,
```

Comment:
Is there a compelling reason not to do it in `create_host_config`?

It looks like the fallback check there is for `network_mode is None`. This will be false if `network_mode` is the empty string. So it seems like the behaviour of `start` and `create_host_config` will differ when the `network_mode` kwarg is omitted:

- `client.start(container)` will start a container with a `network_mode` of `default`.
- `create_host_config()` will return a host config object without a `NetworkMode` key.

If `NetworkMode` is now *required* in host config objects, I feel `create_host_config` should always set it.